passive/Server Header Disclosure.js

```diff
@@ -15,9 +16,31 @@ function scan(ps, msg, src)
     var url = msg.getRequestHeader().getURI().toString();
     var headers = msg.getResponseHeader().getHeaders("Server")
     
-    if (headers != null)
+    if (headers != null && containsPotentialServer(headers))
     {
-        ps.raiseAlert(alertRisk, alertConfidence, alertTitle, alertDesc, url, '', '', '', alertSolution,headers, cweId, wascId, msg);
+        var headers_string = headers.toString();
+        ps.raiseAlert(alertRisk, alertConfidence, alertTitle, alertDesc, url, '', '', '', alertSolution, headers_string, cweId, wascId, msg);
     }
     
 }
+
+function containsPotentialServer(content) {
+
+    var ext = new RegExp("(?:\\d+\\.)+\\d+");
+
+    try {
+        var res = ext.exec(content);
+        if (res == null || res.join('') === ""){
+            return false;
+        }
+
+        else {
+            return true;        
+        }
+
+    }
+
+    catch (err) {
+        return false;
+    }
```

Comment:
As previously mentioned the else isn't necessary.

```js
function containsPotentialServer(content) {

    var ext = new RegExp("(?:\\d+\\.)+\\d+");

    try {
        var res = ext.exec(content);
        if (res == null || res.join('') === "") {
            return false;
        }
        return true;        
     }
    catch (err) {
        return false;
    }
```

Reply:
Removed it.